.github/workflows/pull_request.yml

```
@@ -0,0 +1,18 @@
+name: Code quality
+
+on:
+  push:
+  pull_request:
```

to not run if something changed which doesnt get linted: https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#example-excluding-paths
```suggestion
  pull_request:
    paths-ignore:
      - '**.md'
      - '**.mdx'
      - '**.json'
      - '**.yaml'
```